modules/imgproc/src/median_blur.simd.hpp

```diff
@@ -489,7 +490,149 @@ medianBlur_8u_Om( const Mat& _src, Mat& _dst, int m )
 #undef N
 #undef UPDATE_ACC
 }
+#endif
+
+// Binary search to find the median offset in the histogram
+static inline void
+get_median_ofs256(int &ofs, const uint16_t *hist, const int halfsum) {
```

Comment:
as we discussed, this function should be completely revised, because it's the main bottleneck, especially when the kernel size is not huge. This kernel is assumed to be used for all kernels starting from 7x7. Prepare separate versions for SIMD128, SIMD256 and SIMD_SCALABLE. As in the case of remap patch by Yuantao, put CV_SIMD256 _before_ CV_SIMD128 in order to really enable it, because on Intel in AVX2 mode both CV_SIMD256 and CV_SIMD128 are defined.